### PR TITLE
fix doc for the KeyToBinaryVectorMappingEstimator

### LINKED ISSUE #3472: Issues with KeyToBinaryVectorMappingEstimator
https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.ml.conversionsextensionscatalog?view=ml-dotnet&branch=master

It doesn't shows up as method on `ConversionsExtensionsCatalog` https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.ml.conversionsextensionscatalog?view=ml-dotnet&branch=master because it's part of `ConversionsCatalog`
which I found weird.

https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.ml.conversionscatalog.mapkeytobinaryvector?view=ml-dotnet&branch=master#Microsoft_ML_ConversionsCatalog_MapKeyToBinaryVector_Microsoft_ML_TransformsCatalog_ConversionTransforms_System_String_System_String_

>Convert the key types back to binary vector.

Would be nice to have better explanation.

No table with input/output types.

no examples...

### PR DESCRIPTION
**Pull request by abgoswam**
Fixes #3472 

- Added `Estimator Characteristics` to the `KeyToVectorMappingEstimator` class

### CODE REVIEW

File: src/Microsoft.ML.Transforms/KeyToVectorMapping.cs [hunk 1]
```diff
@@ -426,8 +426,20 @@ private void EncodeValueToBinary(BufferBuilder<float> bldr, uint value, int bits
...
     /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    ///
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | [key](xref:Microsoft.Ml.Data.KeyDataViewType) |
+    /// | Output column data type | A vector of [System.Single](xref:System.Single). |
```
Comment (shmoradims) [inline src/Microsoft.ML.Transforms/KeyToVectorMapping.cs]: >[System.Single](xref:System.Single) [](start = 48, length = 35)

<xref:System.Single>

If the anchor text is the same as the class/type name, you don't really need to use [anchor](xref:UID) pattern. You can save some time by using autolinks: <xref:UID>

**Comment by sfilipi** [comment]
>         ///  [!code-csharp[MapKeyToVector](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Conversions/MapKeyToBinaryVector.cs)]

please Conversions ->Conversion

---
Refers to: src/Microsoft.ML.Transforms/ConversionsCatalog.cs:41 in b770ddd. [](commit_id = b770dddafb8e55a240de90b78dca2ee1fdcab0f4, deletion_comment = False)

File: src/Microsoft.ML.Transforms/KeyToVectorMapping.cs [hunk 2]
```diff
@@ -426,8 +426,21 @@ private void EncodeValueToBinary(BufferBuilder<float> bldr, uint value, int bits
     }
 
     /// <summary>
-    ///  Converts the key types back to binary vectors.
+    /// Estimator for <see cref="KeyToBinaryVectorMappingTransformer"/>. Converts key types to their corresponding binary representation.
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/KeyToVectorMapping.cs]: >to their corresponding binary representation [](start = 92, length = 44)

to the binary representation of the original value.

File: src/Microsoft.ML.Transforms/ConversionsCatalog.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ public static class ConversionsCatalog
         }
 
         /// <summary>
-        ///  Convert the key types back to binary vector.
+        /// Create a <see cref="KeyToBinaryVectorMappingEstimator"/>, which converts key types to their corresponding binary representation.
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/ConversionsCatalog.cs]: >o their corresponding binary representation [](start = 96, length = 43)

binary representation of the original value

**Comment by Ivanidzo4ka** [comment]
>     ///  Converts the key types back to binary vectors.

update it to the pattern we use for other transforms. basically we want circular links. 

---
Refers to: src/Microsoft.ML.Transforms/KeyToVectorMapping.cs:31 in db10158. [](commit_id = db101586c90f3f399c38f1beed2638dd5e203948, deletion_comment = False)

File: src/Microsoft.ML.Transforms/ConversionsCatalog.cs [hunk 4]
```diff
@@ -30,15 +30,15 @@ public static class ConversionsCatalog
         }
 
         /// <summary>
-        ///  Convert the key types back to binary vector.
+        /// Create a <see cref="KeyToBinaryVectorMappingEstimator"/>, which converts key types to their corresponding binary representation of the original value.
         /// </summary>
         /// <param name="catalog">The categorical transform's catalog.</param>
         /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
         /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/ConversionsCatalog.cs]: update to include resulting data types. see other functions for example.

**Comment by abgoswam** [comment]
>     ///  Converts the key types back to binary vectors.

fixed

---
In reply to: [486021436](https://github.com/dotnet/machinelearning/pull/3536#issuecomment-486021436) [](ancestors = 486021436)

---
Refers to: src/Microsoft.ML.Transforms/KeyToVectorMapping.cs:31 in db10158. [](commit_id = db101586c90f3f399c38f1beed2638dd5e203948, deletion_comment = False)

File: src/Microsoft.ML.Transforms/KeyToVectorMapping.cs [hunk 5]
```diff
@@ -426,8 +426,21 @@ private void EncodeValueToBinary(BufferBuilder<float> bldr, uint value, int bits
...
+    /// Estimator for <see cref="KeyToBinaryVectorMappingTransformer"/>. Converts key types to their corresponding binary representation of the original value.
     /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    ///
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | [key](xref:Microsoft.Ml.Data.KeyDataViewType) |
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/KeyToVectorMapping.cs]: >[key] [](start = 35, length = 5)

Scalar of known-size vector of

File: src/Microsoft.ML.Transforms/KeyToVectorMapping.cs [hunk 6]
```diff
@@ -426,8 +426,21 @@ private void EncodeValueToBinary(BufferBuilder<float> bldr, uint value, int bits
...
+    /// | Output column data type | A vector of [System.Single](xref:System.Single). |
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/KeyToVectorMapping.cs]: >vector [](start = 38, length = 6)

Knonw-size vector

File: src/Microsoft.ML.Transforms/ConversionsCatalog.cs [hunk 7]
```diff
@@ -30,15 +30,17 @@ public static class ConversionsCatalog
...
-        ///  Convert the key types back to binary vector.
+        /// Create a <see cref="KeyToBinaryVectorMappingEstimator"/>, which converts key types to their corresponding binary representation of the original value.
         /// </summary>
         /// <param name="catalog">The categorical transform's catalog.</param>
-        /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
-        /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
+        /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.
+        /// The data type is a vector of <see cref="System.Single"/> representing the input value.</param>
+        /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.
+        /// This transform operates over keys.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/ConversionsCatalog.cs]: >keys [](start = 41, length = 4)

scalar of known-size vector of keys

File: src/Microsoft.ML.Transforms/ConversionsCatalog.cs [hunk 8]
```diff
@@ -30,15 +30,17 @@ public static class ConversionsCatalog
...
 
         /// <summary>
-        ///  Convert the key types back to binary vector.
+        /// Create a <see cref="KeyToBinaryVectorMappingEstimator"/>, which converts key types to their corresponding binary representation of the original value.
         /// </summary>
         /// <param name="catalog">The categorical transform's catalog.</param>
-        /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
-        /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
+        /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.
+        /// The data type is a vector of <see cref="System.Single"/> representing the input value.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/ConversionsCatalog.cs]: >vector [](start = 31, length = 6)

known-size vector

**Pull request merged**